File: src/main/kotlin/com/easy/springboot/demo_spring_mvc/exception/NoPermissionException.java
package com.easy.springboot.demo_spring_mvc.exception;

public class NoPermissionException extends Exception {
    String msg;
    Throwable t;

    public NoPermissionException(String msg, Throwable t) {
        this.msg = msg;
        this.t = t;
    }

    public NoPermissionException(String message, String msg, Throwable t) {
        super(message);
        this.msg = msg;
        this.t = t;
    }

    public NoPermissionException(String message, Throwable cause, String msg, Throwable t) {
        super(message, cause);
        this.msg = msg;
        this.t = t;
    }

    public NoPermissionException(Throwable cause, String msg, Throwable t) {
        super(cause);
        this.msg = msg;
        this.t = t;
    }

    public NoPermissionException(String message, Throwable cause, boolean enableSuppression, boolean writableStackTrace, String msg, Throwable t) {
        super(message, cause, enableSuppression, writableStackTrace);
        this.msg = msg;
        this.t = t;
    }
}

File: src/main/kotlin/com/easy/springboot/demo_spring_mvc/exception/NoPermissionException.kt
package com.easy.springboot.demo_spring_mvc.exception

class NoPermissionException : Exception {
    lateinit var msg: String
    lateinit var t: Throwable



    constructor(msg: String, t: Throwable) : super() {
        this.msg = msg
        this.t = t
    }

    constructor(message: String?, msg: String, t: Throwable) : super(message) {
        this.msg = msg
        this.t = t
    }

    constructor(message: String?, cause: Throwable?, msg: String, t: Throwable) : super(message, cause) {
        this.msg = msg
        this.t = t
    }

    constructor(cause: Throwable?, msg: String, t: Throwable) : super(cause) {
        this.msg = msg
        this.t = t
    }

    constructor(message: String?, cause: Throwable?, enableSuppression: Boolean, writableStackTrace: Boolean, msg: String, t: Throwable) : super(message, cause, enableSuppression, writableStackTrace) {
        this.msg = msg
        this.t = t
    }

    constructor(msg: String) : super() {
        this.msg = msg
    }

    constructor(message: String?, msg: String) : super(message) {
        this.msg = msg
    }

    constructor(message: String?, cause: Throwable?, msg: String) : super(message, cause) {
        this.msg = msg
    }

    constructor(message: String?, cause: Throwable?, enableSuppression: Boolean, writableStackTrace: Boolean, msg: String) : super(message, cause, enableSuppression, writableStackTrace) {
        this.msg = msg
    }
}
